src/core/vectortile/qgsmapboxglstyleconverter.cpp

```diff
@@ -2902,14 +2967,22 @@ QString QgsMapBoxGlStyleConverter::retrieveSpriteAsBase64( const QVariant &value
 
 QString QgsMapBoxGlStyleConverter::parseValue( const QVariant &value, QgsMapBoxGlStyleConversionContext &context )
 {
+  QColor c;
+  QString debug;
   switch ( value.type() )
   {
     case QVariant::List:
     case QVariant::StringList:
       return parseExpression( value.toList(), context );
 
+    case QVariant::Bool:
     case QVariant::String:
-      return QgsExpression::quotedValue( value.toString() );
+      c = parseColor( value, context );
```

Comment:
this looks potentially dangerous -- you could have a string value like 'red' misinterpreted as a color value.

Reply:
That's a good point. I've moved the string to color conversion out of the parseValue function. If  parseExpression might expect a color value, it converts the value to a QVariant with type QColor such that parseValue() can handle it properly